packages/calypso-products/src/plans-utilities.ts

```diff
@@ -41,10 +41,40 @@ export function getTermDuration( term: string ): number | undefined {
 
 export const redirectCheckoutToWpAdmin = (): boolean => !! JETPACK_REDIRECT_CHECKOUT_TO_WPADMIN;
 
-export const is2023PricingGridEnabled = () => {
-	const supportedLocales = [ 'en', 'en-gb', 'es' ];
-	return (
-		isEnabled( 'onboarding/2023-pricing-grid' ) &&
-		supportedLocales.includes( getLocaleSlug() ?? '' )
-	);
+/**
+ * Returns if the 2023 Pricing grid optimized design should be visible or not
+ *
+ */
+export const is2023PricingGridEnabled = (): boolean => {
+	const isFeatureFlagEnabled = (): boolean => isEnabled( 'onboarding/2023-pricing-grid' );
+
+	const isLocaleSupported = (): boolean => {
+		const supportedLocales = [ 'en', 'en-gb', 'es' ];
+		return supportedLocales.includes( getLocaleSlug() ?? '' );
+	};
+
+	const isActivePage = (): boolean => {
+		let currentRoutePath = '';
+		if ( typeof window === 'object' && window.location ) {
+			currentRoutePath = window.location.pathname;
+		}
```

Comment:
would this be the same as `window?.location?.pathname ?? ''`

p.s. if we were in all-React mode/usage, then we'd probably be looking at a different implementation without relying explicitly on `window` (we have state selectors for grabbing the current path from context)

Reply:
Actually the `window` object needs to be declared for optional chaining to work, but we can probably add optional chaining to the rest 👍

Reply:
On second thought I will use globalThis!

Reply:
> Ok I switched to typescript global 😇

This is no longer applicable 😬